File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Assorted/UnknownReferenceDeactivationTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o.Tests.Common.Assorted;

namespace Db4objects.Db4o.Tests.Common.Assorted
{
	public class UnknownReferenceDeactivationTestCase : AbstractDb4oTestCase
	{
		public class Item
		{
			public string value;

			public Item(string value)
			{
				this.value = value;
			}
		}

		public virtual void Test()
		{
			UnknownReferenceDeactivationTestCase.Item item = new UnknownReferenceDeactivationTestCase.Item
				("my string");
			Db().Deactivate(item, int.MaxValue);
			Assert.AreEqual("my string", item.value);
		}
	}
}

File: db4oj.tests/src/com/db4o/db4ounit/common/assorted/UnknownReferenceDeactivationTestCase.java
package com.db4o.db4ounit.common.assorted;

import db4ounit.*;
import db4ounit.extensions.*;

public class UnknownReferenceDeactivationTestCase extends AbstractDb4oTestCase {
	
	public static class Item {
		
		public String value;

		public Item(String value) {
	        this.value = value;
        }
		
	}
	
	public void test() {
		final Item item = new Item("my string");
		db().deactivate(item, Integer.MAX_VALUE);
		Assert.areEqual("my string", item.value);
	}

}
